### PR TITLE
Updated statsd client to allow less than 1 ms timing

### PR DESCRIPTION
**Pull request by scbunn**
Currently working on a project that requires the timing of functions that take less than 1 ms to run.  Updated to allow for floating point values so timings of less than 1 ms can be represented.

### CODE REVIEW

File: statsd/client.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def timer(self, stat, rate=1):
 
     def timing(self, stat, delta, rate=1):
         """Send new timing information. `delta` is in milliseconds."""
-        self._send_stat(stat, '%d|ms' % delta, rate)
+        self._send_stat(stat, '%f|ms' % delta, rate)
```
Comment (jsocol) [inline statsd/client.py]: Would it be worth being more specific about the format here, to make the output more predictable, especially in the tests?

File: statsd/tests.py [hunk 2]
```diff
@@ -868,21 +868,20 @@ def test_pipeline_negative_absolute_gauge_tcp():
 
 def _test_big_numbers(cl, proto):
     num = 1234568901234
-    result = 'foo:1234568901234|%s'
     tests = (
         # Explicitly create strings so we avoid the bug we're trying to test.
-        ('gauge', 'g'),
-        ('incr', 'c'),
-        ('timing', 'ms'),
+        ('gauge', 'g', 'foo:1234568901234|%s'),
```
Comment (jsocol) [inline statsd/tests.py]: might as well just drop the `suffix` part and make the `result` the complete thing (along with adjusting the `_check` method to skip the `result % suffix` line, which is the only place `suffix` gets used.

**Comment by jsocol** [comment]
Made a couple of comments. Would you mind hitting those, adding a line to the changelog under 3.2, and squashing it all into one commit? Thanks!

**Comment by scbunn** [comment]
@jsocol Awesome.  I wanted to get to this yesterday, but ran out of time.  I will get the updates made.  How do you prefer squashing commits?

**Comment by jsocol** [comment]
@scbunn All squashed down into one commit, so code and tests changes are together, is perfect. Thanks so much!

File: statsd/client.py [hunk 3]
```diff
@@ -50,7 +50,7 @@ def stop(self, send=True):
         if self._start_time is None:
             raise RuntimeError('Timer has not started.')
         dt = time.time() - self._start_time
-        self.ms = int(round(1000 * dt))  # Convert to milliseconds.
+        self.ms = float(1000 * dt)  # Convert to milliseconds.
```
Comment (jstasiak) [inline statsd/client.py]: @scbunn The ``float()`` call is not necessary, ``dt`` is already a float and ``1000 * dt`` will be a float too.

**Comment by jstasiak** [comment]
This would be useful to have, I'm happy to apply the required changes so that this can be merged.

**Comment by scbunn** [comment]
@jstasiak cool.  I have been wanted to get back to this but have been too busy with other work projects.  If you could wrap this up it would be appreciated.

**Comment by jstasiak** [comment]
@scbunn @jsocol Please see https://github.com/smarkets/pystatsd/commit/a00122498d44c8af450c55b34eb71a1c3a82c930 (``feature/allow_float_timings`` branch in https://github.com/smarkets/pystatsd)

One thing I left unchanged is the ``%f`` format as I don't know what to restrict it to and we can easily change it later.

**Comment by scbunn** [comment]
@jstasiak your branch looks good to me.  As far as %f formatter I would think that not limiting it would be best.  Its possible that somebody might be timing something that needs a very high precision.

**Comment by jstasiak** [comment]
@jsocol Is there a chance this is merged soon? Thanks in advance.

**Comment by jsocol** [comment]
@jstasiak Can you open a PR from the smarkets branch? Thanks for addressing the comments there. The only thing I'd ask is if we can specify the decimals on the float formatter, something like 4 or 6 decimal points, because I don't know that that behavior is guaranteed not to change on a given platform or Python version.

**Comment by jstasiak** [comment]
@jsocol I just created https://github.com/jsocol/pystatsd/pull/66. I can't really spent much time on this right now but feel free to change the float formatting as needed (6 decimal places sounds reasonable).

**Comment by jsocol** [comment]
Thanks @jstasiak, closing this in favor of #66.

**Pull request closed without merging**